// @umijs/preset-dumi/lib/themes/default/Navbar.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _umi = require("umi");

var _LocaleSelect = _interopRequireDefault(require("./LocaleSelect"));

var _context = _interopRequireDefault(require("./context"));

require("./Navbar.less");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var Navbar = function Navbar(_ref) {
  var onMobileMenuClick = _ref.onMobileMenuClick,
      navPrefix = _ref.navPrefix;

  var _useContext = (0, _react.useContext)(_context.default),
      locale = _useContext.locale,
      rootPath = _useContext.rootPath,
      mode = _useContext.mode,
      title = _useContext.title,
      logo = _useContext.logo,
      locales = _useContext.locales,
      navs = _useContext.navs;

  return /*#__PURE__*/_react.default.createElement("div", {
    className: "__dumi-default-navbar",
    "data-mode": mode
  }, /*#__PURE__*/_react.default.createElement("button", {
    className: "__dumi-default-navbar-toggle",
    onClick: onMobileMenuClick
  }), /*#__PURE__*/_react.default.createElement(_umi.Link, {
    className: "__dumi-default-navbar-logo",
    style: {
      backgroundImage: logo && "url('".concat(logo, "')")
    },
    to: rootPath
  }, title), /*#__PURE__*/_react.default.createElement("nav", null, navPrefix, navs.map(function (nav) {
    return /^\/[\w-]/.test(nav.path) ? /*#__PURE__*/_react.default.createElement(_umi.NavLink, {
      to: nav.path,
      key: nav.path
    }, nav.title) : /*#__PURE__*/_react.default.createElement("a", {
      target: "_blank",
      href: nav.path,
      key: nav.path
    }, nav.title, /*#__PURE__*/_react.default.createElement("svg", {
      xmlns: "http://www.w3.org/2000/svg",
      "aria-hidden": "true",
      x: "0px",
      y: "0px",
      viewBox: "0 0 100 100",
      width: "15",
      height: "15",
      className: "__dumi-default-external-link-icon"
    }, /*#__PURE__*/_react.default.createElement("path", {
      fill: "currentColor",
      d: "M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"
    }), /*#__PURE__*/_react.default.createElement("polygon", {
      fill: "currentColor",
      points: "45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"
    })));
  }), /*#__PURE__*/_react.default.createElement(_LocaleSelect.default, null)));
};

var _default = Navbar;
exports.default = _default;
