// @umijs/preset-dumi/lib/transformer/demo/options.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getBabelOptions = void 0;

function _react() {
  const data = _interopRequireDefault(require("react"));

  _react = function _react() {
    return data;
  };

  return data;
}

var _context = _interopRequireDefault(require("../../context"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const getBabelOptions = ({
  isTSX,
  fileAbsPath
}) => {
  var _ctx$umi, _ctx$umi$config, _ctx$umi2, _ctx$umi2$config;

  return {
    filename: fileAbsPath,
    presets: [[require.resolve('@umijs/babel-preset-umi/app'), {
      reactRequire: false,
      typescript: isTSX
    }], ...(((_ctx$umi = _context.default.umi) === null || _ctx$umi === void 0 ? void 0 : (_ctx$umi$config = _ctx$umi.config) === null || _ctx$umi$config === void 0 ? void 0 : _ctx$umi$config.extraBabelPresets) || [])],
    plugins: [[require.resolve('@babel/plugin-transform-modules-commonjs'), {
      strict: true
    }], ...(((_ctx$umi2 = _context.default.umi) === null || _ctx$umi2 === void 0 ? void 0 : (_ctx$umi2$config = _ctx$umi2.config) === null || _ctx$umi2$config === void 0 ? void 0 : _ctx$umi2$config.extraBabelPlugins) || [])],
    ast: true,
    babelrc: false,
    configFile: false
  };
};

exports.getBabelOptions = getBabelOptions;
